dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/FilteredCollectionsReportUtils.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.utils;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.solr.client.solrj.SolrServerException;
+import org.dspace.app.rest.contentreports.Filter;
+import org.dspace.app.rest.model.FilteredCollectionRest;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.Item;
+import org.dspace.content.service.CollectionService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * This is the Service dealing with the {@link FilteredCollectionRest} logic.
+ *
+ * @author Jean-François Morin (Université Laval)
+ */
+@Component
+public class FilteredCollectionsReportUtils {
```

Comment:
Again, it'd be useful to add tests to verify the functionality in this class...and ensure it continues to work correctly in the future.